### PR TITLE
[iOS] Support Achievement Rally with NFC

### PR DESCRIPTION
**Pull request by ry-itto**
## Issue
- close #ISSUE_NUMBER

## Overview (Required)
- Support Achievement Rally on iOS
  - with NFC
  - with DeepLink

## Links
- 

## Screenshot (Optional if screenshot test is present or unrelated to UI)
Before | After
:--: | :--:
 | 

## Movie (Optional)
Before | After
:--: | :--:
<video src="" width="300" > | <video src="" width="300" >

Recording of "Electriic Eel" achievement.

https://github.com/DroidKaigi/conference-app-2023/assets/30540303/f8fb499b-8f9f-4a86-a5e2-a8fe74097215

### CODE REVIEW

**Comment by woxtu** [comment]
Awesome!

File: app-ios/Modules/Sources/Achievements/AchievementsView.swift [hunk 1]
```diff
@@ -1,29 +1,121 @@
...
+                                AchievementImage(
+                                    target: Achievement.electriceel,
+                                    savedAchievements: state.achievements,
+                                    activeImage: Assets.Images.achievementElectricEelActive,
+                                    inactiveImage: Assets.Images.achievementElectricEel
+                                )
+                            }
+                            Button {
+                                Task {
+                                    let result = await viewModel.read()
```
Comment (tkhs0604) [inline app-ios/Modules/Sources/Achievements/AchievementsView.swift:64]: Oh, we need an action just before reading NFC tags 👀

**Review by tkhs0604 — APPROVED** [review]
LGTM!

**Comment by ry-itto** [comment]
Sorry I have some work left to do, so I'll make it a draft.

**Pull request merged**